mtags/src/main/scala-3/scala/meta/internal/pc/CompletionValue.scala

```diff
@@ -1,58 +1,86 @@
 import dotty.tools.dotc.core.Flags.*
 import dotty.tools.dotc.core.Symbols.NoSymbol
 import dotty.tools.dotc.core.Symbols.Symbol
 import dotty.tools.dotc.interactive.Completion
+import dotty.tools.dotc.transform.SymUtils.*
+import dotty.tools.dotc.util.ParsedComment
 import org.eclipse.lsp4j.CompletionItemKind
+import org.eclipse.lsp4j.CompletionItemTag
+
+sealed trait CompletionValue:
+  def label: String
+
+  final def completionItemKind(using Context): CompletionItemKind =
+    this match
+      case _: CompletionValue.Keyword => CompletionItemKind.Keyword
+      case _: CompletionValue.NamedArg => CompletionItemKind.Field
+      case v: (CompletionValue.Compiler | CompletionValue.Workspace |
+            CompletionValue.Scope) =>
+        val symbol = v.symbol
+        if symbol.is(Package) || symbol.is(Module) then
+          // No CompletionItemKind.Package (https://github.com/Microsoft/language-server-protocol/issues/155)
+          CompletionItemKind.Module
+        else if symbol.isConstructor then CompletionItemKind.Constructor
+        else if symbol.isClass then CompletionItemKind.Class
+        else if symbol.is(Mutable) then CompletionItemKind.Variable
+        else if symbol.is(Method) then CompletionItemKind.Method
+        else CompletionItemKind.Field
+  end completionItemKind
+
+  final def documentation(using Context): Option[String] =
+    forSymOnly(ParsedComment.docOf(_).map(_.renderAsMarkdown), None)
+
+  final def lspTags(using Context): List[CompletionItemTag] =
+    forSymOnly(
+      sym =>
+        if sym.isDeprecated then List(CompletionItemTag.Deprecated) else Nil,
+      Nil
+    )
+
+  private def forSymOnly[A](f: Symbol => A, orElse: => A): A =
+    this match
+      case CompletionValue.SymbolOnly(sym) => f(sym)
+      case _ => orElse
 
-case class CompletionValue(
-    label: String,
-    symbol: Symbol,
-    kind: Kind,
-    isCustom: Boolean = false,
-    insertText: Option[String] = None
-):
-
-  def completionItemKind(using ctx: Context): CompletionItemKind =
-    if kind == CompletionValue.Kind.Keyword then CompletionItemKind.Keyword
-    else if symbol.is(Package) || symbol.is(Module) then
-      CompletionItemKind.Module // No CompletionItemKind.Package (https://github.com/Microsoft/language-server-protocol/issues/155)
-    else if symbol.isConstructor then CompletionItemKind.Constructor
-    else if symbol.isClass then CompletionItemKind.Class
-    else if symbol.is(Mutable) then CompletionItemKind.Variable
-    else if symbol.is(Method) then CompletionItemKind.Method
-    else CompletionItemKind.Field
+  def anySymbol: Option[Symbol] =
+    this match
+      case CompletionValue.SymbolOnly(sym) => Some(sym)
+      case CompletionValue.NamedArg(_, sym) => Some(sym)
+      case _ => None
 
 end CompletionValue
 
 object CompletionValue:
+  sealed trait SymbolOnly extends CompletionValue:
+    def symbol: Symbol
+  object SymbolOnly:
+    def unapply(v: CompletionValue): Option[Symbol] =
```

Comment:
It's needed to distinct usual symbol completions from named-arg one.
It would result to that there would be two methods that is more confusing that current approach:
```
def symbol: Option[Symbol] // for symbolOnly
def anySymbol: Option[Symbol] // for symbolOnly and named arg
```

Reply:
Why would we need both in this situation? Why not just `def symbol: Option[Symbol] ` ?

Reply:
Since anySymbol is only used in one place, maybe we could havae `def symbol: Option[Symbol] ` to be more general, but instead of `anySymbol` we could do `def description(history: ShortenedNames, printer: SymbolPrinter)` ?

Reply:
changed it a bit - deleted `anySymbol` and added `def description`.
I wanted to do that at the beginning but it fells like there is a need to combine  `SymbolPrinter` with `ShortenedNames` to simplify method calls. But that is thing is for separate PR

Reply:
That's what I also started thinking :sweat_smile:

Reply:
Do we need the unapply still? We can match on `Symbolic`

Reply:
I've tried to drop it but it's still looks a bit better that match by type in a few places.
It's tiny one so I think it won't be a problem to keep it.

Reply:
It does box it additionally and since we use it in sorting I think it might be better to drop the unapply.